File: src/main/java/com/aladdinworks9/service/MaintenanceScheduleService.java
package com.aladdinworks9.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks9.domain.MaintenanceSchedule;
import com.aladdinworks9.dto.MaintenanceScheduleDTO;
import com.aladdinworks9.dto.MaintenanceScheduleSearchDTO;
import com.aladdinworks9.dto.MaintenanceSchedulePageDTO;
import com.aladdinworks9.dto.MaintenanceScheduleConvertCriteriaDTO;
import com.aladdinworks9.service.GenericService;
import com.aladdinworks9.dto.common.RequestDTO;
import com.aladdinworks9.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface MaintenanceScheduleService extends GenericService<MaintenanceSchedule, Integer> {

	List<MaintenanceSchedule> findAll();

	ResultDTO addMaintenanceSchedule(MaintenanceScheduleDTO maintenanceScheduleDTO, RequestDTO requestDTO);

	ResultDTO updateMaintenanceSchedule(MaintenanceScheduleDTO maintenanceScheduleDTO, RequestDTO requestDTO);

    Page<MaintenanceSchedule> getAllMaintenanceSchedules(Pageable pageable);

    Page<MaintenanceSchedule> getAllMaintenanceSchedules(Specification<MaintenanceSchedule> spec, Pageable pageable);

	ResponseEntity<MaintenanceSchedulePageDTO> getMaintenanceSchedules(MaintenanceScheduleSearchDTO maintenanceScheduleSearchDTO);
	
	List<MaintenanceScheduleDTO> convertMaintenanceSchedulesToMaintenanceScheduleDTOs(List<MaintenanceSchedule> maintenanceSchedules, MaintenanceScheduleConvertCriteriaDTO convertCriteria);

	MaintenanceScheduleDTO getMaintenanceScheduleDTOById(Integer maintenanceScheduleId);







}

File: aladdinworks9_ui/src/services/MaintenanceScheduleService.js
import http from "../http-common"; 

class MaintenanceScheduleService {
  getAllMaintenanceSchedules(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/maintenanceSchedule/maintenanceSchedules`, searchDTO);
  }

  get(maintenanceScheduleId) {
    return this.getRequest(`/maintenanceSchedule/${maintenanceScheduleId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/maintenanceSchedule?field=${matchData}`, null);
  }

  addMaintenanceSchedule(data) {
    return http.post("/maintenanceSchedule/addMaintenanceSchedule", data);
  }

  update(data) {
  	return http.post("/maintenanceSchedule/updateMaintenanceSchedule", data);
  }
  
  uploadImage(data,maintenanceScheduleId) {
  	return http.postForm("/maintenanceSchedule/uploadImage/"+maintenanceScheduleId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new MaintenanceScheduleService();
